PowerShell Contributing.md

```diff
@@ -0,0 +1,479 @@
+
+- To stash changes, go `git stash`
+- To checkout master, do `git checkout master`
+
+Now we need to check for changes and pull them in.
+
+- Tell git to find out if there are changes in any of its remotes: `git fetch --all -p`
+- Tell git to rebase your local repo on the 'upstream' remote (replace the name 'upstream' if necessary):
+`git rebase upstream/master`
+
+Rebasing is the way to tell git to replay your work on top of the other changes that were merged to master in the mean time. So now we have pulled in all changes from the original repo into our clone locally. These changes need to be pushed to your GitHub fork.
+
+- Tell git to push local changes to origin: `git push` or `git push origin/master`.
+
+For more detailed instructions with screenshots, see [here](https://info.sapien.com/index.php/version-control/github-how-to-update-your-fork).
+
+### Submitting your contribution 
+
+Do small git commits while you work, and don't forget to git push the work on your branch every once in a while.
+When you are satisfied with your work, you can create a Pull Request.
+You can create a PR from your own GitHub on the branch you created for you work on this.
+It helps to reference your original Issue where you asked for guidance what to work on, use a '#' sign and put the number of your Issue behind it.
+Fill any fields the PR requests from you.
+
+Once you have created a PR, the automated CI process will start to run.
+It will run ALL tests, on Windows, Linux and OSX.
+You can follow the status of the running tests in your PR.
+It takes around 20 minutes or so to finish.
+Every time you push a new commit to your branch that you created the PR from, it will trigger a new CI run.
+
+If you want to signal the maintainers that you are working on stuff, but are not ready yet, you can rename your PR to start with `WIP:` for Work In Progress.
+
+Every Pull Request in the PowerShell repo gets an assignee.
+A person who you can ask for help when something is not working properly (sometimes CI can be glitchy and fail on something else than your code), or if you just have a question.
+Be patient - half of the work on the PowerShell repository is being done by volunteers, members of the community like you who do this in their spare time.
+And staff members of the PowerShell Team have all kinds of tasks assigned to them.
+
+Your work in the PR will be reviewed by two maintainers.
+They will ask you to change things here and there, maybe even about code you did not touch, _because, just like you, they love PowerShell_ and they want the best for the product.
+If you don't know what they mean, just say so, no worries.
+They are all very helpful and friendly people.
+It can take a few weeks for the whole review/finetuning process to complete.
+
+Once all the CI checks turn green, and your reviewers are satisfied, your PR will be merged and your code is added to the codebase :-)
```

Comment:
Might be a good idea to note that releases are done roughly every month or two, so it may be some weeks before merged code shows up in a public release version. 🙂